coding/logging.rst

```diff
@@ -0,0 +1,231 @@
+#######
+Logging
+#######
+
+This page provides guidance to developers for using logging with :lmod:`lsst.log`  in the science pipeline code base. 
+Developers are encouraged to insert log messages whenever and wherever they might be useful, with appropriate component names and levels.
+
+.. _logger-names:
+
+Logger Names
+============
+Logger names should generally start with the fully qualified name of the module/file containing the logger, without the leading ``lsst.``.
+If the logger is saved as a variable in a class, it is often appropriate to name the logger after the class.
+Logger names use ``.`` as component separators, not ``::``, even in C++.
+Some example logger names are ``afw.image.MaskedImage``, ``meas.algorithms.starSelector``, etc.
+
+
+Basic Usage in Python
+=====================
+
+The basic Python interface of :lmod:`lsst.log` is made to be somewhat similar to Python logging.
+The simplest way to log is:
+
+.. code-block:: python
+
+   import lsst.log
+   lsst.log.info('Some information during normal operation')
+   lsst.log.warn('Here is a warning!')
+
+
+The example above logs to the root logger.
+A better practice is to use a named logger following our :ref:`name convention <logger-names>`, and log to that logger in your code to indicate where the logging messages originate.
+For example,
+
+.. code-block:: python
+
+   logger = lsst.log.Log.getLogger("meas.algorithms.starSelector")
+   logger.info("This is information about the star selector algorithm execution. %f", 3.14)
+
+In Python, comma-separated arguments are preferred over the string formatting so it is done only if the log record is to be printed.
```

Comment:
I didn't quite understand this sentence. Are you saying not to use `%` formatting, and that `lsst.log` will do the string formatting for you? I.e., is `so it is done only if the log record is to be printed` an action taken by the developer or by `lsst.log` internally?